solutions/src/main/java/ru/sberbank/school/task13/cache/CacheProxy.java

```diff
@@ -0,0 +1,334 @@
+
+/**
+ * Created by Mart
+ * 10.07.2019
+ **/
+public class CacheProxy {
+    private String directoryPath;
+
+    public CacheProxy(String directoryPath) {
+        String message = "path to directory can not be ";
+        Objects.requireNonNull(directoryPath,message + "null");
+        if (directoryPath.isEmpty()) {
+            throw new IllegalArgumentException(message + "empty");
+        }
+
+        File directory = new File(directoryPath);
+        this.directoryPath = getCheckedPath(directory, directoryPath);
+    }
+
+    private static String getCheckedPath(File directory, String directoryPath) {
+        String path;
+        if (directory.exists() && directory.isDirectory()) {
+            path = directoryPath;
+        } else {
+            boolean created = directory.mkdir();
+            if (created) {
+                path = directoryPath;
+            } else {
+                throw new IllegalArgumentException("error! directory can not be created");
+            }
+        }
+        return path;
+    }
+
+    public Object cache(Object service) {
+        Class c1ass = service.getClass();
+        return Proxy.newProxyInstance(c1ass.getClassLoader(),
+                c1ass.getInterfaces(),
+                new CacheProxyHandler(service));
+    }
+
+    private class CacheProxyHandler implements InvocationHandler {
+        private Object service;
+        private Map<String, Object> cacheMap;
+
+        CacheProxyHandler(Object service) {
+            this.service = service;
+            this.cacheMap = new HashMap<>();
+        }
+
+        @Override
+        public Object invoke(Object proxy, Method method, Object[] args)
+                throws InvocationTargetException, IllegalAccessException {
+
+            Object result = null;
+            Cache cacheAnnotation = method.getAnnotation(Cache.class);
+
+            if (!handlerInvokeMethodArgsCheck(cacheAnnotation, method, args)) {
+                return method.invoke(service, args);
+            }
+
+            CacheType cacheType = cacheAnnotation.cacheType();
+
+            if (cacheType.equals(CacheType.JVM)) {
+                result = invokeJVM(method, args);
+            } else if (cacheType.equals(CacheType.FILE)) {
+                result = invokeFile(method, args);
+            } else {
+                throw new CacheProxyException("unknown Cache type is declared in " + method.getName());
+            }
+
+            return result;
+        }
+
+        private boolean handlerInvokeMethodArgsCheck(Annotation cacheAnnotation, Method method, Object[] args) {
+            if (cacheAnnotation == null) {
+                return false;
+            }
+
+            if (method.getReturnType().equals(void.class)) {
+                throw new IllegalArgumentException("called method can not be void");
+            }
+
+            return true;
+        }
+
+        private Object invokeJVM(Method method, Object[] args) throws InvocationTargetException,
```

Comment:
Abbreviation in name &#39;invokeJVM&#39; must contain no more than &#39;2&#39; consecutive capital letters.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-1529733848>*

Reply:
invokeJvm